packages/ai-bot/main.ts

```diff
@@ -370,30 +370,72 @@ Common issues are:
             );
           }
         })
-        .on('functionCall', async (functionCall) => {
-          console.log('Function call', functionCall);
-          let args;
-          try {
-            args = JSON.parse(functionCall.arguments);
-          } catch (error) {
-            Sentry.captureException(error);
-            return await sendError(
-              client,
-              room,
-              error,
-              initialMessage.event_id,
+        .on('message', async (message) => {
+          // if role is assistant
+          if (message.role === 'assistant') {
+            //send each tool call
+            log.info(
+              'Tool Calls',
+              JSON.stringify(message.tool_calls || [], null, 2),
             );
+            for (let toolCall of message.tool_calls || []) {
+              log.info('Tool Call', toolCall);
+              sentCommands += 1;
+              let functionCall = toolCall.function;
+              let args;
+              try {
+                args = JSON.parse(functionCall.arguments);
+              } catch (error) {
+                Sentry.captureException(error);
+                return await sendError(
+                  client,
+                  room,
+                  error,
+                  initialMessage.event_id,
+                );
+              }
+              if (functionCall.name === 'patchCard') {
+                sentCommands += 1;
+                return await sendOption(
+                  client,
+                  room,
+                  args,
+                  initialMessage.event_id,
+                );
+              } else {
+                log.info('sending option #', sentCommands, functionCall);
+                const body =
+                  args['description'] ||
+                  `Calling tool ${functionCall.name}(${functionCall.arguments})`;
+                let messageObject = {
+                  body: body,
+                  msgtype: 'org.boxel.command',
+                  formatted_body: body,
+                  format: 'org.matrix.custom.html',
+                  data: {
+                    functionCall: toolCall,
```

Comment:
This is only required if we want to send results back to openai from the tool calls.